modules/library/main/src/test/java/org/geotools/filter/visitor/SimplifyingFilterVisitorTest.java

```diff
@@ -224,8 +228,8 @@ public void testNegateBetween() {
         Literal l20 = ff.literal(20);
         Filter f = ff.not(ff.between(prop, l10, l20));
         Filter result = (Filter) f.accept(visitor, null);
-        assertEquals(ff.or(Arrays.asList((Filter) ff.less(prop, l10), ff.greater(prop, l20))),
-                result);
+        // *not* simplified
+        assertEquals(f, result);
```

Comment:
The first translation you made is wrong, let's do it step by step:
not(any-between(10, 20))  -> (expand between)
not(and(any-greater-equal(10), any-less-equal(20)) -> (apply De Morgan)
or(not(any-greater-equal(10), not(any-less-equal(20)) -> (apply your comparison inversions)
or(all-less-than(10), all-greater-than(20))

If we go back now to his example, this translation seems to work, any-between(10,20) returns true, negated to false.
all-less-than(10) is false, and all-greater-than(20) is false too, thus false, thus equivalent at least for this example.

However, it does not take much to prepare an example where the equivalence is not met, if the multi-valued property has values [5,25] then not(any-between(10,20)) return true, but or(all-less-than(10), all-greater-than(20)) still returns false.

This also has a severe implication, because the original filter could have been just have expressed as not(and(any-greater-equal(10), any-less-equal(20)), which
is something anyone can enter, no? At which point did we lose the equivalence?

For the [5,25] case:

not(any-between(10, 20))  -> true
not(and(any-greater-equal(10), any-less-equal(20)) -> true
or(not(any-greater-equal(10), not(any-less-equal(20)) -> true
or(all-less-than(10), all-greater-than(20)) -> false

And well, the issue is in the last step, if we just consider the inversion of not(any-greater-equal(10)) -> all-less-than(10) you can see that just negating the match-action does not do you any good, it is indeed not equivalent.

Long story short, the logic inversion here should just not be applied when we are dealing with a complex feature type, or we don't know the target feature type (well actually, if one wanted, a check of whether the properties involved are multi-valued or not could be performed, but I'll leave that as an excercise the reader, app-schema has much more serious issues than some filter simplification right now).